product_state/models/product_template.py

```diff
@@ -35,22 +35,26 @@ def _inverse_product_state_id(self):
 
     @api.model
     def _get_default_product_state_id(self):
-        return self.env["product.state"].search([("default", "=", True)], limit=1).id
+        return self.env["product.state"].search([("default", "=", True)], limit=1)
 
     @api.depends("product_state_id")
     def _compute_product_state(self):
         for product_tmpl in self:
             product_tmpl.state = product_tmpl.product_state_id.code
 
     def _inverse_product_state(self):
-        ProductState = self.env["product.state"]
         for product_tmpl in self:
-            product_state = ProductState.search(
-                [("code", "=", product_tmpl.state)], limit=1
+            self._set_product_state_id(product_tmpl)
+
+    def _set_product_state_id(self, record):
```

Comment:
```suggestion
            product_tmpl._set_product_state_id()

    def _set_product_state_id(self):
```

Reply:
The method is being used by both `product.template` and `product.product` so `record` is needed. It is now promoted to `api.model` level.